op-acceptance-tests/tests/base/deposit_test.go

```diff
@@ -0,0 +1,96 @@
+import (
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/ethereum-optimism/optimism/op-devstack/devtest"
+	"github.com/ethereum-optimism/optimism/op-devstack/dsl"
+	"github.com/ethereum-optimism/optimism/op-devstack/presets"
+	"github.com/ethereum-optimism/optimism/op-devstack/stack/match"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/lmittmann/w3"
+)
+
+func TestL1ToL2Deposit(gt *testing.T) {
+	// Create a test environment using op-devstack
+	t := devtest.SerialT(gt)
+	sys := presets.NewMinimal(t)
+
+	// Wait for L1 node to be responsive
+	l1Client := sys.L1Network.Escape().L1ELNode(match.FirstL1EL).EthClient()
+	require.Eventually(t, func() bool {
+		_, err := l1Client.ChainID(t.Ctx())
+		return err == nil
+	}, 60*time.Second, 500*time.Millisecond, "L1 node not responsive")
+
+	// Now wait for a block
+	sys.L1Network.WaitForBlock()
+
+	// Fund Alice on L1
+	fundingAmount := eth.ThousandEther
+	alice := sys.FunderL1.NewFundedEOA(fundingAmount)
+	alice.VerifyBalanceExact(fundingAmount)
+
+	alicel2 := dsl.NewEOA(alice.Key(), sys.L2EL)
+
+	// Get the optimism portal address
+	rollupConfig := sys.L2Chain.Escape().RollupConfig()
+	portalAddr := rollupConfig.DepositContractAddress
+
+	depositAmount := eth.OneEther
+
+	// Define the deposit function and encode arguments
+	funcDeposit := w3.MustNewFunc(`depositTransaction(address,uint256,uint64,bool,bytes)`, "")
+	args, err := funcDeposit.EncodeArgs(
+		alice.Address(),       // _to
+		depositAmount.ToBig(), // _value
+		uint64(300_000),       // _gasLimit
+		false,                 // _isCreation
+		[]byte{},              // _data
+	)
+	require.NoError(t, err)
+
+	// Prepare the transaction
+	nonce, err := l1Client.PendingNonceAt(t.Ctx(), alice.Address())
+	require.NoError(t, err)
+	gasPrice, err := l1Client.SuggestGasPrice(t.Ctx())
+	require.NoError(t, err)
+	chainID, err := l1Client.ChainID(t.Ctx())
+	require.NoError(t, err)
+
+	tx := types.NewTransaction(
+		nonce,
+		portalAddr,
+		depositAmount.ToBig(), // ETH to deposit
+		500_000,               // L1 gas limit
+		gasPrice,
+		args,
+	)
+
+	// Sign and send
+	signedTx, err := types.SignTx(tx, types.NewEIP155Signer(chainID), alice.Key().PrivateKey())
+	require.NoError(t, err)
+	err = l1Client.SendTransaction(t.Ctx(), signedTx)
+	require.NoError(t, err)
+
+	// Wait for finalization
+	sys.L1Network.WaitForFinalization()
+
+	receipt, err := l1Client.TransactionReceipt(t.Ctx(), signedTx.Hash())
+	require.NoError(t, err)
+	require.Equal(t, uint64(1), receipt.Status, "deposit tx failed")
+
+	// Verify the deposit was successful
+	gasCost := new(big.Int).Mul(new(big.Int).SetUint64(receipt.GasUsed), gasPrice)
+	expectedFinalL1 := new(big.Int).Sub(fundingAmount.ToBig(), depositAmount.ToBig())
+	expectedFinalL1.Sub(expectedFinalL1, gasCost)
+
+	alice.VerifyBalanceExact(eth.WeiBig(expectedFinalL1))
+	alicel2.VerifyBalanceExact(depositAmount)
```

Comment:
```suggestion
	// Sequencer takes a while to confirm data from L1.
	t.Require().Eventually(func() bool {
		head := sys.L2CL.HeadBlockRef(types.LocalUnsafe)
		return head.L1Origin.Number >= receipt.BlockNumber.Uint64()
	}, time.Second*10, time.Second, "awaiting deposit to be processed by L2")
	alicel2.VerifyBalanceExact(depositAmount)
```

It's unsafe to wait for balance to just arrive on L2. The deposit from L1 is not instantaneously processed. You should await the L1 origin confirmation.